tests/rados/test_upgrade_warn.py

```diff
@@ -0,0 +1,86 @@
+"""
+Module to check for the warning "OSD_UPGRADE_FINISHED" during upgrades
+
+"""
+
+import datetime
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.ceph_admin.orch import Orch
+from ceph.rados.core_workflows import RadosOrchestrator
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Test to check if the warning "OSD_UPGRADE_FINISHED" is generated when "require_osd_release" does ot match
+    the current release during upgrades.
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm_obj = CephAdmin(cluster=ceph_cluster, **config)
+    cluster_obj = Orch(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm_obj)
+
+    log.debug("Starting upgrade")
+    try:
+        cluster_obj.set_tool_repo()
+        time.sleep(5)
+        cluster_obj.install()
+        time.sleep(5)
+
+        # Check service versions vs available and target containers
+        cluster_obj.upgrade_check(image=config.get("container_image"))
+
+        installer = ceph_cluster.get_nodes(role="installer")[0]
+        base_cmd = "sudo cephadm shell -- ceph"
+        ceph_version, err = installer.exec_command(cmd=f"{base_cmd} version")
```

Comment:
I believe this is legacy code and could be changed to -
`ceph_version, err = cephadm_obj.shell(["ceph version"])`
Very minor though

Reply:
Updated.